### ISSUE TITLE
Deferring collection allocation and clearing builders

### ISSUE THREAD

**Issue by pushbit** [immutables/immutables#329]
When clearing a `jdkOnly` builder, collections are cleared.

```java
	@Immutable
	@Style(jdkOnly = true, clearBuilder = true)
	abstract class Baz {
		abstract List<Object> baz();
	}
```
generates
```java
    public Builder clear() {
      this.baz.clear();
      return this;
    }
```

But when also deferring collection allocation, collections are set to null.

```java
	@Immutable
	@Style(jdkOnly = true, clearBuilder = true, deferCollectionAllocation = true)
	abstract class Baz {
		abstract List<Object> baz();
	}
```
generates
```java
    public Builder clear() {
      this.baz = null;
      return this;
    }
```

To get the best of both worlds in this case, maybe we could use the below instead?

```java
      if (this.baz != null) {
        this.baz.clear();
      }
```

Edit: This could be useful for `Modifiable` as well.

**Comment by elucash** [comment]
this may be a nice touch, I'll add it in next release, thank you for suggestion!

**Comment by elucash** [comment]
Released in 2.2.1 Thank you!
